pyam/core.py

```diff
@@ -171,17 +171,18 @@ def _init(self, data, meta=None, index=DEFAULT_META_INDEX, **kwargs):
                     "Initializing from list is not supported, "
                     "use `IamDataFrame.append()` or `pyam.concat()`"
                 )
+
             # read from file
             try:
                 data = Path(data)  # casting str or LocalPath to Path
-                if data.is_file():
-                    logger.info(f"Reading file {data}")
-                    _data = read_file(data, index=index, **kwargs)
-                else:
-                    raise FileNotFoundError(f"File {data} does not exist")
             except TypeError:  # `data` cannot be cast to Path
-                msg = "IamDataFrame constructor not properly called!"
-                raise ValueError(msg)
+                raise ValueError("IamDataFrame constructor not properly called!")
```

Comment:
This error is (conceptually) at the end of all options: if it's not a pandas.DataFrame or an ixmp.Scenario or something file-like, then the constructor was not properly called.

Reply:
Ok I see what you mean. From my point of view it was just that the `ValueError` came as part of a `try except` that only tried to cast data into a `pathlib.Path` object. For a user I think it might be more helpful to directly tell them that after checking if their input is a pandas data frame or series, ixmp TimeSeries, etc... the last thing that was tried was the conversion into a path to read data and that that failed too.

Reply:
Thanks for the nudge, I restructured this section of the code to have a more obvious structure. Not sure if there are other types than string, `LocalPath` or `Path` that should be treated as "file-like object".

Reply:
Thanks for the restructure, I think those three classes are fine, maybe `os.path` as well but other than that I don't think we're missing anything.
Out of curiosity what's `LocalPath`? I've never seen it, what is it used for?

Reply:
Oh right I see, thanks for the explanation. Looks like a good way to solve that.